app/workers/segment_delete_user_worker.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class SegmentDeleteUserWorker < ActiveJob::Base # rubocop:disable Rails/ApplicationJob # That would be Rails 5 :)
+  class MissingUserIDError < StandardError; end
+
+  def perform(event_id)
+    event = EventStore::Repository.find_event!(event_id)
+    raise(MissingUserIDError) unless (user_id = event.data[:user_id])
+    SegmentDeleteService.new(user_id).call!
+  rescue ActiveRecord::RecordNotFound, MissingUserIDError => exception
```

Comment:
Avoid raising and rescueing in the same method/class. That reeks (normally)  :)

Reply:
Just for the record, for everyone to know, I told Hery:

> For the worker in particular i think that i don't have a choice 🤔 if you have a better suggestion for that case, let me know plz

And he told me that he does and that we will speak about it later 😉

Reply:
Spoken with Hery. I should remove from here `rescue` and even this `raise` because it will raise an error and inform anyway.

Reply:
This is raised by the event not been in the database.

In the first place, it should not happen. If it does raise an error Sidekiq will retry and it will be worn off after 3 attempts. Sure that is 3 more attempts but that is fine as it should not happen.

Reply:
> This is raised by the event not been in the database.

This is referring to `ActiveRecord::RecordNotFound`. But we spoke about all of it and the conclusion is https://github.com/3scale/porta/pull/717#discussion_r271664315